provider/server/machine_create_test.go

```diff
@@ -55,12 +55,9 @@ var _ = Describe("CreateMachine", func() {
 		))
 
 		DeferCleanup(func(ctx SpecContext) {
+			_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
```

Comment:
i think tests can be unstable by this way. Secure solution is wrapping  DeleteMachine in own Eventually which will try call DeleteMachine maybe max 10second with 1second period, if DeleteMacine return error.

Root cause is place of calling function. Ideally DeferCleanup will left environment in the same state before tests were started.

Reply:
how were testes written never failed for DeleteMachine? i checked code and some edge cases can be occurred (unmarshall, unmarshall, write problem, wrong open file permissions, ...).  You can get error from machine store. They will be extremely rare and  many of them cannot be fix automatically probably (some can be) but implement secure solution isn't so hard and we can very easy improve stability of tests.
So pls add this additional Eventually blocks. thx

Reply:
We have already secured solution by handling error from `DeleteMachine`. Adding it into Eventually block does not make any difference for `DeleteMachine` call, if it fail once then it will fail again also. That's why this issue has been raised to avoid multiple calls to `DeleteMachine`. If `DeleteMachine` fails somehow then genuinely there will be some error which Eventually() block will also not recover like you mentioned unmarshall, wrong file perms, these are non-recoverable by Eventually().

Reply:
I had discussion with @so-sahu and we agreed that if this occurs in practical scenario then we will have separate PR for this.

Reply:
i will implement it.